Source/Tools/FEXLoader/FEXLoader.cpp

```diff
@@ -151,47 +157,42 @@ void InterpreterHandler(fextl::string* Filename, const fextl::string& RootFS, fe
   // Is the file large enough for shebang
   if (ReadSize <= 2) {
     close(FD);
-    return;
+    return false;
   }
 
   // Handle shebang files
   if (Data[0] == '#' && Data[1] == '!') {
-    fextl::string InterpreterLine {Data.begin() + 2, // strip off "#!" prefix
-                                   std::find(Data.begin(), Data.end(), '\n')};
-    fextl::vector<fextl::string> ShebangArguments {};
+    std::string_view InterpreterLine {Data.begin() + 2, // strip off "#!" prefix
+                                      std::find(Data.begin(), Data.end(), '\n')};
+    fextl::vector<std::string_view> ShebangArguments {};
 
     // Shebang line can have a single argument
-    fextl::istringstream InterpreterSS(InterpreterLine);
-    fextl::string Argument;
-    while (std::getline(InterpreterSS, Argument, ' ')) {
-      if (Argument.empty()) {
-        continue;
+    auto Begin = InterpreterLine.begin();
+    auto ArgEnd = Begin;
+    auto End = InterpreterLine.end();
+    while (ArgEnd != End && Begin != End) {
+      // The end of an argument ends with a space or the end of the interpreter line.
+      ArgEnd = std::find(Begin, End, ' ');
+      if (Begin == ArgEnd) {
```

Comment:
Won't this erroneously return immediately for inputs like `#! /usr/bin/env sh`?

I think changing to `ArgEnd == End` and moving it in front of the `Begin = ArgEnd + 1;` is correct here. The `ArgEnd != End` can then be removed from the loop condition, and `ArgEnd` doesn't need to be declared outside of the loop.

Reply:
Changed this around to live in FHU so we can unittest the thing. It now gets properly unittested and works as expected.